### ISSUE TITLE
typo in obtheory

### ISSUE THREAD

**Issue by band** [orcmid/miser#16]
In Section 1. Logical Notation

I think "p ⇔ q read p if-and-only-if q, true when true both are true or"
should be
p ⇔ q read p if-and-only-if q, true when both are true or"

("true when true both" has an extra "true" ?)

**Comment by orcmid** [comment]
The web page was updated on 2018-01-18 to display obtheory.txt 0.2.5 notation definitions having the correcction.

**Comment by band** [comment]
Another typo in obtheory (Q: shall I just put all typos into this issue?):

Lines 33,34:
     * There are two selector functions, ob.a(z) and ob.b(z) that given
any ob and determine an ob as their result.

"given any ob determine an ob ...." -- the "and" does not belong there.

**Comment by orcmid** [comment]
> Another typo in obtheory (Q: shall I just put all typos into this issue?):

Is in ob.txt, not obtheory.txt.  I moved it to new open issue #19.

Thank you for these -- they become invisible to the author :)

**Comment by band** [comment]
Doh! Sorry about the mixup; I am switching back and forth between the two documents.

**Comment by band** [comment]
Another typo? I'm just reading ahead and came across this in lines 228-229 (Section 4. Notes and References):

Representations of strings that have
other strings as their beads was also introduced by Doug Ross and

should "beads" be "heads"?

**Comment by orcmid** [comment]
> Another typo? I'm just reading ahead and came across this in lines 228-229 (Section 4. Notes and References):
> Representations of strings that have
> other strings as their beads was also introduced by Doug Ross and
> should "beads" be "heads"?

Nope, beads.  Think about a string data format in which any bead of the string is a single character or is a string imbedded (not concatenated) at that point.  The head of a such a string is a single string bead, and that may be either a single character or an embedded-string.

A list structure is easier to think of in this regard.  Think of a linked list in which any element of a non-empty list is a literal symbol or the enclosure of such a list structure.  :)

That's the counterpart oMiser idiom for nesting strings as beads within a superior string.

**Comment by orcmid** [comment]
@band I just noticed a difference between head/tail and first/rest.  In the normal formulation of strings, the head of a string is another string, as is the tail of the string.  First/Rest has first be the first element (e.g., a character) and Rest is of the same type (i.e, a list form).  Some systems require that all elements of a list be of the same type and I suppose one can interpret a string (typically) as a list of characters or an equivalent array/vector.  This gets into (intended) interpretation and representation goodies.

**Comment by band** [comment]
OK. I had an intuition that I needed to read more before saying "typo?".

I was also thinking about Lisp atoms and lists after reading your comment. And I agree that the intended use of a list for representation is an interesting question. Jumping up several levels, the idea of using a relational database to house the information in an address book is also a representation question. Is this right? Is "I have atoms and lists, but want to, say, parse, English sentences" also in this representation and interpretation arena?

----
Also, I do see that pairing with ob.c allows for collections of mixed <ob>'s  in one unit.

**Comment by orcmid** [comment]
> ... Jumping up several levels, the idea of using a relational database to house the information in an address book is also a representation question. Is this right? Is "I have atoms and lists, but want to, say, parse, English sentences" also in this representation and interpretation arena?

The TL;DR: Yes

It is turtles everywhere.

As @band pointed out in a phone conversation, what gets nailed down to create stable levels involves ontological commitment.  We grant being to some level, have no doubt of it, and then move upward in the creation of stable tiers.  I find that (apart from probably over-use of paradigm) that the treatment in [PLato and the Nerd](http://platoandthenerd.org/index.html) is very useful in this regard.
